File: uniffi/src/bindings/python/templates/TopLevelFunctionTemplate.py
{%- match func.return_type() -%}
{%- when Some with (return_type) %}

def {{ func.name()|fn_name_py }}({%- call py::arg_list_decl(func.arguments()) -%}):
    {%- call py::coerce_args(func.arguments()) %}
    _retval = {% call py::to_rs_call(func.ffi_func()) %}
    return {{ "_retval"|lift_py(return_type) }}

{% when None -%}

def {{ func.name()|fn_name_py }}({%- call py::arg_list_decl(func.arguments()) -%}):
    {%- call py::coerce_args(func.arguments()) %}
    {% call py::to_rs_call(func.ffi_func()) %}
{% endmatch %}

File: uniffi/src/bindings/kotlin/templates/TopLevelFunctionTemplate.kt
{%- match func.return_type() -%}
{%- when Some with (return_type) %}

fun {{ func.name()|fn_name_kt }}({%- call kt::arg_list_decl(func) -%}): {{ return_type|type_kt }} {
    val _retval = {% call kt::to_rs_call(func) %}
    return {{ "_retval"|lift_kt(return_type) }}
}

{% when None -%}

fun {{ func.name()|fn_name_kt }}({% call kt::arg_list_decl(func) %}) =
    {% call kt::to_rs_call(func) %}
{% endmatch %}
